metadata/src/main/java/org/apache/kafka/metalog/MetaLogLeader.java

```diff
@@ -26,7 +26,7 @@
     private final int nodeId;
     private final long epoch;
 
-    MetaLogLeader(int nodeId, long epoch) {
+    public MetaLogLeader(int nodeId, long epoch) {
```

Comment:
@cmccabe Not sure if this needs to be package private. I needed to make it public for testing in `BrokerMetadataListenerTest` which is not in the same package.

Reply:
> Not sure if this needs to be package private. I needed to make it public for testing in BrokerMetadataListenerTest which is not in the same package.

You can make it `private[kafka]` (pretty darn close to public, of course, but not quite)

Reply:
oh, `private[kafka]` doesn't exist in Java, unfortunately.  I'm OK with making it public for now.